### PR TITLE
[LMS-79][Integration] Implement Course Create

### PR DESCRIPTION
**Pull request by ragg09-sunasterisk**
## Issue Link
[LMS-79 [INTEGRATION] Implement Course Create](https://framgiaph.backlog.com/view/LMS-79)

## Defintion of Done
- should be able create a new course
- redirect to the details page
- has validation
- alert using toastify
- Fix folder structure

## Steps to reproduce
make sure that you create a .env.local in client
`cp client/.env.example client/.env.local`

NEXT_PUBLIC_ANALYTICS_API_BASE_URL=http://localhost:8000/api/

- install the following
`npm i axios`
`npm i react-toastify`
- fill out the form in this  [page](http://localhost:3000/trainer/course/create)

### be
 - pipfile
 - settings.py
### fe
 - course/create
 - course/<id>
 - demo/components/
 - InputField Component
 - Select Component
 - TextField Component

## Test Cases
- try to submit without data
- try with data

## Notes
- created_by has default user for now, since the only ID we can use is the admin

## Screenshots
image
image

course_create
course_create

### CODE REVIEW

File: api/app/settings.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'rest_framework',
-    
+    'corsheaders',
```
Comment (kdacudag) [inline api/app/settings.py:48]: Install this in Pipfile

**Review by mikorosales — COMMENTED** [review]
I tried running using docker ster, category list is not rendered

image

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [x] able to create a new course.
- [x] redirect to the details page.
- [x] Tested creating without data and it did not go thru, also tested selecting each category and it all went thru.
- [x] course created is saved in the database.
- [x]  No ESLint warnings or errors.
- [ ] No error found, when running npm run build.

Comments: 2
Test time: 55 mins
Iterations: 1

Proof:
create

**Review by mikorosales — APPROVED** [review]
image
8a5b40977f990bb6e0a5d0ac29a6d6c0

**Comment by escbooster12-dev** [comment]
- follow exact design
image

**Comment by ragg09-sunasterisk** [comment]
Updated Design
image

**Comment by mikorosales** [comment]
Suggestion:
Ster is it possible to add nalang din a label input for course category field?

> Updated Design image

Suggestion:
Ster Rene, just a suggestion, is it possible to add nalang din a label for course category? Just for uniformity of the form inputs

**Comment by Marielleregner7** [comment]
asas
I think its better po na tanggaling yung "Select" na word 🙇

**Review by mikorosales — APPROVED** [review]
Approved!
- Can add course
- Design match
- No lint errors

**Review by Marielleregner7 — APPROVED** [review]
APPROVED
- [x] able to create a new course.
- [x]  redirect to the details page.
- [x]  Tested creating without data and it did not go thru, also tested selecting each category and it all went thru.
- [x]  course created is saved in the database.
- [ ]  No ESLint warnings or errors.
- [ ]  No error found, when running npm run build.

Comments: 9
Test time: 69 mins
Iterations: 5

Proof:
createcourse1

**Review by escbooster12-dev — APPROVED** [review]

**Pull request merged**